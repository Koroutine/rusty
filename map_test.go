package rusty

import (
	"testing"

	. "github.com/smartystreets/goconvey/convey"
)

func TestGet(t *testing.T) {

	target := map[string]interface{}{
		"a": 1,
		"b": map[string]interface{}{
			"c": "2",
			"d": map[string]interface{}{
				"e": 3,
				"f": true,
			},
		},
	}

	Convey("Find value at path", t, func() {
		value := Get[string](target, "a.b.c").Unwrap()

		So(value, ShouldEqual, "2")
	})

	Convey("Find int value at path", t, func() {
		value := Get[int](target, "a.b.c.d.e").Unwrap()

		So(value, ShouldEqual, 3)
	})

	Convey("Default value if path not found", t, func() {
		value := Get[string](target, "a.b.c.z").UnwrapOr("test")

		So(value, ShouldEqual, "test")
	})

	Convey("Unwrap panics if path not found", t, func() {
		So(func() { Get[string](target, "a.b.c.z").Unwrap() }, ShouldPanic)
	})

	Convey("Unwrap panics if type assertion fails", t, func() {
		So(func() { Get[int](target, "a.b.c").Unwrap() }, ShouldPanic)
	})

}
