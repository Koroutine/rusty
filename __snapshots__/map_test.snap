
[TestSet - 1]
{
 "a": 1,
 "b": {
  "c": "2",
  "d": {
   "e": 3,
   "f": true,
   "g": "test"
  }
 }
}
---

[TestSet - 2]
{
 "a": 1,
 "b": {
  "c": "2",
  "d": {
   "e": 3,
   "f": true,
   "g": "hello",
   "r": "test"
  }
 }
}
---

[TestSet - 3]
{
 "a": 1,
 "arr": [
  {
   "key": "test"
  }
 ]
}
---

[TestSet - 4]
{
 "Records": [
  {
   "Outcome_details": [
    {
     "id": "test",
     "obj": {
      "a": "test3",
      "b": "test4"
     }
    }
   ],
   "Record_details": {
    "id": "test",
    "obj": {
     "a": "test1",
     "b": "test2"
    }
   }
  }
 ]
}
---

[TestSet - 5]
{
 "a": 1,
 "b": {
  "c": "2",
  "d": {
   "e": 3,
   "f": true,
   "g": "hello",
   "g_r": "test"
  }
 }
}
---
